jetstream/analysis.py

```diff
@@ -202,7 +203,36 @@ def calculate_metrics(
                 enrollments_table,
             )
 
-            self.bigquery.execute(metrics_sql, res_table_name)
+            if "main" in metrics_sql:
+                # todo: remove/undo
+                # Run metrics queries in shared-prod as a workaround
+                shared_prod_client = BigQueryClient(
+                    project="moz-fx-data-shared-prod", dataset="tmp"
+                )
+
+                tmp_enrollments_table = f"moz-fx-data-shared-prod.tmp.{enrollments_table}"
+
+                try:
+                    shared_prod_client.client.copy_table(
+                        f"{self.project}.{self.dataset}.{enrollments_table}", tmp_enrollments_table
+                    ).result()
+                except Conflict:
+                    pass
```

Comment:
And here as well, come to think of it -- the enrollments table might have changed if we add a segment or something, so we shouldn't assume the `tmp` version is up to date. It should be small enough that copying unconditionally won't be a huge expense.

Reply:
Yeah, using `WRITE_TRUNCATE` when copying should fix this